src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -346,20 +405,23 @@ protected override async Task DeleteMessagesToAsync(string persistenceId, long t
 
             // only update the sequence number of the top of the journal
             // is about to be deleted.
-            if (highestSeqNo <= toSequenceNr) 
+            if (highestSeqNo <= toSequenceNr)
             {
                 await SetHighSequenceId(persistenceId, highestSeqNo);
             }
+
             if (_settings.Transaction)
             {
                 var sessionOptions = new ClientSessionOptions { };
-                using (var session = await _journalCollection.Value.Database.Client.StartSessionAsync(sessionOptions/*, cancellationToken*/))
+                using (var session =
+                       await _journalCollection.Value.Database.Client.StartSessionAsync(
+                           sessionOptions /*, cancellationToken*/))
```

Comment:
Didn't use CTS on the session here - did in other places. Do it everywhere or do it JUST on the transaction commits?

Reply:
Resolved - using it here now per @Arkatufus feedback